desc/objectives/utils.py

```diff
@@ -130,32 +132,49 @@ def factorize_linear_constraints(objective, constraint):  # noqa: C901
             )
         A = A[unfixed_rows][:, unfixed_idx]
         b = b[unfixed_rows]
+
+    # unscaled particular solution to get scale of x
     unfixed_idx = indices_idx
     if A.size:
-        Ainv_full, Z = svd_inv_null(A)
+        A_inv, Z = svd_inv_null(A)
     else:
-        Ainv_full = A.T
+        A_inv = A.T
         Z = np.eye(A.shape[1])
-    Ainv_full = jnp.asarray(Ainv_full)
-    Z = jnp.asarray(Z)
-    b = jnp.asarray(b)
-    xp = put(xp, unfixed_idx, Ainv_full @ b)
+    xp = put(xp, unfixed_idx, A_inv @ b)
+    x_scale = np.where(np.abs(xp) < 1, 1, np.abs(xp))  # TODO: adjust threshold?
```

Comment:
Right now, this helps rescale variables with very large amplitudes. We could change this threshold to something like 
```
x_scale = np.where(np.abs(xp) < 1e-6, 1, np.abs(xp))
```
to also help rescale variables with very small amplitudes. Is that worth doing? I don't think it can hurt, but also I don't think we have any variables (yet) with small normalizations.